server/modules/routing/readwritesplit/readwritesplit.hh

```diff
@@ -166,8 +186,11 @@ struct Config
         connection_keepalive(config_get_integer(params, "connection_keepalive")),
         max_slave_replication_lag(config_get_integer(params, "max_slave_replication_lag")),
         rw_max_slave_conn_percent(0),
-        max_slave_connections(0)
+        max_slave_connections(0),
+        causal_read((causal_read_t)config_get_enum(params, causal_read_key, causal_read_values)),
+        causal_read_timeout(config_get_string(params, causal_read_timeout_key))
     {
+        retry_failed_reads = causal_read != CAUSAL_READ_NON_SUPPORT ? true: false;
```

Comment:
Seems that retrying of reads will always be enabled when causal reads are disabled, even if they are explicitly disabled.

Another question I have is that why is the retrying of reads disabled when causal reads are enabled? Retrying should be safe as the same set of SQL commands will just be executed on another server.